src/pages/workspace/WorkspaceEditorPage.js

```diff
@@ -50,26 +51,35 @@ class WorkspaceEditorPage extends React.Component {
     }
 
     onImageSelected(image) {
+        updateLocalPolicyValues(this.props.policy.id, {isAvatarUploading: true});
         this.setState({previewAvatarURL: image.uri});
 
         // Store the upload avatar promise so we can wait for it to finish before updating the policy
         this.uploadAvatarPromise = uploadAvatar(image).then(url => new Promise((resolve) => {
+            updateLocalPolicyValues(this.props.policy.id, {isAvatarUploading: false});
```

Comment:
We can prevent repeating this line by putting in in an `always()` right?

Reply:
Can you elaborate this please?

Reply:
Got it. Sure. Will update.

Reply:
Done